src/main/java/objects/Player.java

```diff
@@ -63,4 +99,128 @@ public void printStats(){
+
+            //TODO: Make the check for ranged training availability better
+            for (String arrow : new String[]{"bronzeArrow", "ironArrow"}){
+                for (String bow : new String[]{"shortbow", "oakShortbow"}) {
+                    if (Constant.UNLOCKED_ITEM_DATABASE.contains(arrow) && Constant.UNLOCKED_ITEM_DATABASE.contains(bow)) {
+                        trainableStats.put(Skills.Ranged, true);
+                    }
+                }
+            }
+
+            //TODO: Add better checking for rune availability, through 100% drops and rune shops
+            for (String shopName : Constant.CHUNK_DATABASE.getElement(Integer.toString(chunk)).getShops()){
+                for (String item : Constant.SHOP_DATABASE.getElement(shopName).getStock()){
+                    if (item == "mindRune"){
+                        trainableStats.put(Skills.Magic, true);
+                    }
+                }
+            }
+
+
+            //Can every non-combat skill be classified as a process, sometimes without inputs/outputs or both?
+            for (String skillingLocations : Constant.CHUNK_DATABASE.getElement(Integer.toString(chunk)).getSkillingLocations()){
+                for (String process : Constant.PROCESSING_TOOL_DATABASE.getElement(skillingLocations).getProcessesNames()){
+                    if (Constant.PROCESS_DATABASE.getElement(process).isDoable(this)){
+                        trainableStats.put(Constant.PROCESS_DATABASE.getElement(process).getSkillType(), true);
+                    }
+                }
+            }
+        }
+    }
+
+    public EnumMap<Skills, Integer> skillsToNextChunk (){
+        //TODO: loop through skilling nodes to determine levels instead of looking at the chunks
+        EnumMap<Skills, Integer> out = new EnumMap<Skills, Integer>(Skills.class);
+        for (Skills skill : Skills.values()) {
+            out.put(skill, 0);
+        }
+
+        for (int chunk : unlockedChunks){
+            EnumMap<Skills, Integer> chunkSkills = Constant.CHUNK_DATABASE.getElement(Integer.toString(chunk)).getSkillReqs(this);
+            for (Skills skill : Skills.values()){
+                out.put(skill, Math.max(out.get(skill), chunkSkills.get(skill)));
+            }
+        }
+
+        return out;
+    }
+
+    public EnumMap<WeaponType, EnumMap<EquipmentSlot, EquippableItem[]>> BiSItems(){
+        EnumMap<WeaponType, EnumMap<EquipmentSlot, EquippableItem[]>> p = new EnumMap<WeaponType, EnumMap<EquipmentSlot, EquippableItem[]>>(WeaponType.class);
+
+        for (WeaponType type : WeaponType.values()) {
+            EnumMap<EquipmentSlot, EquippableItem[]> BiS = new EnumMap<EquipmentSlot, EquippableItem[]>(EquipmentSlot.class);
+            for (EquipmentSlot slot : EquipmentSlot.values()) {
+                BiS.put(slot, null);
+            }
+
+            for (EquipmentSlot slot : EquipmentSlot.values()) {
+                ArrayList<EquippableItem> BiSArr = new ArrayList<EquippableItem>();
```

Comment:
Avoid capitals for first letter in variables. 
```suggestion
                ArrayList<EquippableItem> bisArr = new ArrayList<EquippableItem>();
```

Reply:
Follow-through renaming